src/utils/MastodonUtils.C

```diff
@@ -124,6 +128,47 @@ MastodonUtils::mean(const std::vector<Real> & vector)
   return sum / vector.size();
 }
 
+std::vector<std::vector<Real>>
+MastodonUtils::mean(const std::vector<std::vector<Real> *> & history_acc,
```

Comment:
If my understanding is correct, then this function should not need the time as input. You can just loop over the rows and do the mean calculation. The "req_size" should just be the length of the first vector and it can be assumed that all the other vectors are this same length.

However, as you are looping you should use a mooseAssert to perform this check.

Also, this is a stand-alone function that should be unit tested; @cbolisetti can help you with this.

Reply:
- Omitted time as an input to the mean function.
- Checked if the vectors of response histories are all of equal length using the MastodonUtils::checkEqualSize function. Implemented mooseError to throw an exception.
- Implemented the unit test, it passed.